lms/djangoapps/student_enrollment/management/commands/enrollment.py

```
@@ -45,48 +48,69 @@ def handle(self, *args, **options):
+
+                # Get the Program that contains the th Zoho program code
+                program = Program.objects.get(
+                    program_code=program_to_enroll_in)
+
+                # Enroll the student in the program
+                program_enrollment_status = program.enroll_student_in_program(
+                    user.email,
+                    exclude_courses=EXCLUDED_FROM_ONBOARDING)
+
+                # Send the email
+                email_sent_status = program.send_email(
+                    user, enrollment_type, password)
+
+                # Set the students access level (i.e. determine whether or
+                # not a student is allowed to access to the LMS.
+                # Deprecated...
+                access, created = ProgramAccessStatus.objects.get_or_create(
+                    user=user, program_access=True)
+
+                if not created:
+                    access.allowed_access = True
+                    access.save()
+
+                post_to_zapier(settings.ZAPIER_ENROLLMENT_URL,
+                               {'email': user.email})
+
+                enrollment_status = EnrollmentStatusHistory(
+                    student=user,
+                    program=program,
+                    registered=bool(user),
+                    enrollment_type=enrollment_type,
+                    enrolled=bool(program_enrollment_status),
+                    email_sent=email_sent_status)
+                enrollment_status.save()
+
+            email_content = ('<h2>Successfully enrolled %d students.</h2>'
+                                    % len(zoho_students))
+            send_success_or_exception_email(email_type='success',
+                                            content=email_content,
+                                            from_address=FROM_ADDRESS,
+                                            to_address=TO_ADDRESS)
+
+        except Exception as exception:
```

Sorry, but I have a very strong aversion to catching `Exception` , and kinda swallowing it - I would ask to reduce the scope of the `try` and to catch only the actual exception class that is raised by `send_success_or_exception_email` (should be [smtplib.SMTPException](https://docs.python.org/3/library/smtplib.html#smtplib.SMTPException) )

---

This is where Sentry (or other logging provider) can be handy. You can do:

try:
    send_mail(...)
except smtplib.SMTPException as se:
    log.exception("There was a problem sending email to %s", to_address)
    raise

So the exception gets re-raised (if that's what we want) and the logged exception gets sent to Sentry for future consumption.

---

@yoniLavi, not sure I understand this change here. We are trying to catch all errors which prevent the enrollment script from running, but this would only catch errors in sending emails, correct? The idea behind this was that we currently do not have anything in place to tell us whether the enrollment script ran or not and we often times get requests from SC asking us this question. In order to understand if it ran or not we then have to ssh into the server and manually check the log and in the CRM as well. We can change it so that the exception is not caught and we only get an email if the script ran successfully, but I thought having the exception in an email as well would be beneficial and allow us to troubleshoot quicker.

---

@yoniLavi and @gomyar, after following up with Nakita there is currently only one real type of exception that we have should expect which is a `ObjectDoesNotExist` exception. Therefore, I changed the enrollment script to catch this exception only. I also added a try...except statement to the method that sends an email in case of an `SMTPException` and am printing the full stacktrace to the log using the `log.exception` statement.